.github/workflows/ci.yml

```diff
@@ -184,115 +184,82 @@ jobs:
-      uses: actions/checkout@v2
+      uses: actions/checkout@v2.3.4
     - name: Set up QEMU
-      id: qemu
+      if: ${{ matrix.qemu }}
       uses: docker/setup-qemu-action@v1
-    - name: Available platforms
-      run: echo ${{ steps.qemu.outputs.platforms }}
-    - name: Setup Python
-      uses: actions/setup-python@v2
-    - name: Install cython
-      if: matrix.no-extensions == ''
-      uses: py-actions/py-dependency-install@v2
       with:
-        path: requirements/cython.txt
-    - name: Cythonize
-      if: matrix.no-extensions == ''
-      run: |
-        make cythonize
-    - name: Install tools
-      run: |
-        docker run --rm -v ${{ github.workspace }}:/ws:rw --workdir=/ws \
-          ${{ env.img }} ${{ env.py }} -m pip install -U setuptools wheel
-    - name: Make wheel
-      run: |
-        docker run --rm -v ${{ github.workspace }}:/ws:rw --workdir=/ws \
-          ${{ env.img }} ${{ env.py }} setup.py bdist_wheel
-    - name: Repair wheel wheel
+        platforms: all
+      id: qemu
+    - name: Prepare emulation
       run: |
-        docker run --rm -v ${{ github.workspace }}:/ws:rw --workdir=/ws \
-          ${{ env.img }} auditwheel repair dist/*.whl --wheel-dir wheelhouse/
-    - name: Upload artifacts
-      uses: actions/upload-artifact@v2
-      with:
-        name: dist
-        path: wheelhouse/*
-
-  build-binary:
-    name: Binary wheels
-    strategy:
-      matrix:
-        pyver: [3.6, 3.7, 3.8, 3.9, 3.10]
-        os: [macos, windows]
-        arch: [x86, x64]
-        exclude:
-        - os: macos
-          arch: x86
-      fail-fast: false
-    runs-on: ${{ matrix.os }}-latest
-    needs: pre-deploy
-    steps:
-    - name: Checkout
-      uses: actions/checkout@v2
+        if [[ -n "${{ matrix.qemu }}" ]]; then
+          # Build emulated architectures only if QEMU is set,
+          # use default "auto" otherwise
+          echo "CIBW_ARCHS_LINUX=${{ matrix.qemu }}" >> $GITHUB_ENV
+        fi
+      shell: bash
     - name: Setup Python
-      uses: actions/setup-python@v2
-      with:
-        python-version: ${{ matrix.pyver }}
-        architecture: ${{ matrix.arch }}
+      uses: actions/setup-python@v2.2.2
     - name: Install cython
-      if: matrix.no-extensions == ''
       uses: py-actions/py-dependency-install@v2
       with:
         path: requirements/cython.txt
     - name: Cythonize
-      if: matrix.no-extensions == ''
       run: |
         make cythonize
-    - name: Install dependencies
-      run: |
-        python -m pip install -U setuptools wheel
-    - name: Make wheel
-      run:
-        python setup.py bdist_wheel
-    - name: Upload artifacts
-      uses: actions/upload-artifact@v2
+    - name: Build wheels
+      uses: pypa/cibuildwheel@v2.2.0a1
+      env:
+        CIBW_ARCHS_MACOS: x86_64 arm64 universal2
+    - uses: actions/upload-artifact@v2
       with:
         name: dist
-        path: dist
+        path: ./wheelhouse/*.whl
 
   deploy:
```

Comment:
I'd also add a `permissions` setting to this job in order to limit the scope of the GH token.

Reply:
Make a PR please